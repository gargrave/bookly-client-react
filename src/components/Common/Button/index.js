import React from 'react'
import { func, string } from 'prop-types'

const classname = (type) => {
  if (type === 'success') {
    return 'buttons-success'
  }
}

const Button = ({
  onClick,
  text,
  type,
}) => {
  return (
    <button
      className={classname(type)}
      style={{ marginLeft: '10px' }}
      onClick={onClick}>
      { text }
    </button>
  )
}

Button.propTypes = {
  onClick: func.isRequired,
  text: string.isRequired,
  type: string.isRequired,
}

export default Button
