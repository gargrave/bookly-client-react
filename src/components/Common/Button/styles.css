.bookly-button {
  margin-left: 10px;
}
