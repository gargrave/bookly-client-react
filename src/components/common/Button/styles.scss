.bookly-button {
  &:not(&--left) {
    margin-left: 10px;
  }
}
