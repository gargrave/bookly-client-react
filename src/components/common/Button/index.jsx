// @flow
import React from 'react';
import { bool, func, oneOf, string } from 'prop-types';

import { buildClasses } from '../../../utils/cssHelpers';

import './styles.css';

type Props = {
  canSubmit?: boolean,
  classes?: string,
  onClick: Function,
  position?: string,
  text: string,
  type: string,
};

const acceptableTypes = [
  'success',
  'secondary',
  'info',
  'warning',
  'danger',
  'light',
  'dark',
];

const acceptablePositions = [
  'left',
];

function buttonClass(type: string) {
  if (acceptableTypes.includes(type)) {
    return `button-${type}`;
  }
  return '';
}

function positionClass(position: string) {
  if (acceptablePositions.includes(position)) {
    return `button-${position}`;
  }
  return '';
}

function buildClassList(
  type: string,
  position: string,
  classes: string,
) {
  return buildClasses(
    ['button', positionClass(position)],
    [buttonClass(type), ...classes.split(' ')]
  );
}

function Button({
  canSubmit,
  classes = '',
  onClick,
  position = '',
  text,
  type,
}: Props) {
  return (
    <button
      type={canSubmit ? 'submit' : 'button'}
      className={buildClassList(type, position, classes)}
      onClick={onClick}>
      { text }
    </button>
  );
}

Button.propTypes = {
  canSubmit: bool,
  classes: string,
  onClick: func.isRequired,
  position: oneOf(acceptablePositions),
  text: string.isRequired,
  type: oneOf(acceptableTypes).isRequired,
};

export default Button;
