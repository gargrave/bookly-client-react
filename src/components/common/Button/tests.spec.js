import React from 'react'
import { shallow } from 'enzyme'

import Button from './'

describe('Button', () => {
  let component
  let props

  beforeEach(() => {
    props = {
      onClick: jest.fn(),
      text: 'ButtonText',
      type: 'success',
    }
  })

  it('matches the snapshot', () => {
    component = shallow(<Button {...props} />)
    expect(component).toMatchSnapshot()
  })

  it('renders as a "success" button', () => {
    component = shallow(<Button {...props} />)
    expect(component.hasClass('bookly-button')).toBeTruthy()
    expect(component.hasClass('button-success')).toBeTruthy()
  })

  it('renders the correct text', () => {
    component = shallow(<Button {...props} />)
    expect(component.text()).toEqual(props.text)
  })

  it('calls the callback as expected')
})
