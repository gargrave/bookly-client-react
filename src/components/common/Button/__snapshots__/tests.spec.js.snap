// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <button
    className="bookly-button button-success"
    onClick={[Function]}
    type="button"
>
    ButtonText
</button>,
  "nodes": Array [
    <button
      className="bookly-button button-success"
      onClick={[Function]}
      type="button"
>
      ButtonText
</button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Button
        onClick={[Function]}
        text="ButtonText"
        type="success"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "onClick": [Function],
          "text": "ButtonText",
          "type": "success",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <button
          className="bookly-button button-success"
          onClick={[Function]}
          type="button"
>
          ButtonText
</button>,
        "_debugID": 2,
        "_renderedOutput": <button
          className="bookly-button button-success"
          onClick={[Function]}
          type="button"
>
          ButtonText
</button>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Button
    onClick={[Function]}
    text="ButtonText"
    type="success"
/>,
}
`;
